### DISCUSSION TITLE
KeycloakClient CRD in the new operator

### DISCUSSION THREAD

**Discussion by stoetti** [keycloak/keycloak#11724] [Archive]
Are there any plans on supporting this CRD which was available in the old operator?

**Comment by jbpin** [comment]
Same. I came here for that. In a gitops manner, it could be good to be able to create client with CRD to include authz configuration and get client credentials from secret.

**Comment by andreaTP** [comment]
Thanks for the interest!
You can already add clients using the RealmImport CRD (along with the enclosing Realm).
Would you mind elaborating on the specific use case if this doesn't fit your needs?

cc. @vmuzikar

**Reply by jbpin** [reply]
RealmImport are operation done once. The realm is configured once. As stated in the documentation : 
>The Realm Import CR only supports creation of new realms and doesn’t updates or delete those.
Changes to the realm performed directly on Keycloak are not synched back in the CR.

In a gitops manner, each application should be able to create its own resources when deployed instead of have configuration done once and for all. 

The previous operator allowed us to deploy client and configuration needed for this apps only in a declarative way.

**Reply by vmuzikar** [reply]
Hello, we certainly do plan to introduce some sort of declarative way to manage Keycloak resources (like Clients, Realms etc.) from K8s – probably introduce new CRs for Realms/Clients/... leveraging static configuration. However, since we rely with this on the new Store.X which is not yet ready, we decided to offer at least a stop-gap solution with the Realm Import CR.

Hope this answers your question and thanks for your patience.

**Reply by stoetti** [reply]
Thank for the heads up.
As our current deployment strategy relies on the KeycloakClient CRD and we are evaluating migrating to the quarkus distribution, do you already know which release will contain the CRDs?

**Comment by bbigras** [comment]
Any update on this?
